gammacat/collection.py

```diff
@@ -258,16 +262,46 @@ def process_datasets(self):
                         dataset.write(path)
 
     def make_index_file_for_input(self):
-        # TODO: this is a temp hack. Fill correctly using GammaCatResourceIndex
-        data = OrderedDict()
-        data['info'] = gammacat_info.info_dict
-        data['data'] = self.input_data.datasets.to_dict()['data']
-        data['files'] = self.config.list_of_files()
-        path = self.config.index_datasets_json
-        write_json(data, path)
+        resources = []
+        for info_filename in self.input_data.info_yaml_list:
+            info_data = load_yaml(info_filename)
+            if info_data['data_entry']['status'] == 'missing':
+                continue
+            # TODO: Decide which datasets are copied to output collection by the keywords in 
+            # 'status' and 'reviewed' in info.yaml
+            # e.g if info_data['data_entry']['status'] == 'complete':
+            for dataset in info_data['datasets']:
+                if dataset.endswith('yaml'):
+                    resource = DataSet.read(info_filename.parent / dataset).resource
+                    resource.location = str(info_filename.parent.relative_to(self.config.in_path) / dataset)
+                    resources.append(resource)
+                elif dataset.endswith('ecsv'):
+                    #TODO: Clean this messy stuff up!
```

Comment:
@pdeiml - Please give it your best shot to clean this up now before we merge. I already gave the suggestion that you should introduce a helper function. In addition, a lot of the lines you duplicate many times can be moved after the if / else.

Reply:
Ok, I will give it a try this afternoon or tomorrow morning.